src/basic/in-addr-util.c

```diff
@@ -464,3 +464,44 @@ int in_addr_mask(int family, union in_addr_union *addr, unsigned char prefixlen)
 
         return -EAFNOSUPPORT;
 }
+
+int in_addr_prefix_from_string(const char *p, int *family, union in_addr_union *prefix, uint8_t *prefixlen) {
+        union in_addr_union buffer;
+        const char *e, *l;
+        uint8_t k;
+        int f, r;
+
+        assert(p);
+
+        /* prefixlen */
+        e = strchr(p, '/');
+        if (e) {
+                r = safe_atou8(e + 1, &k);
+                if (r < 0)
+                        return r;
+
+                l = strndupa(p, e - p);
+        } else
+                l = p;
```

Comment:
if no "/" is found you do not initialize k, that doesn't look right. I figure you should initialize it to 128 resp 32.